Frameworks/include/NSOrderedSetInternal.h

```diff
@@ -0,0 +1,28 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+#pragma once
+
+#import <Starboard.h>
+#import <Foundation/NSOrderedSet.h>
+#import <map>
+#import <vector>
+
+@interface NSOrderedSet () {
+@protected
+    std::vector<StrongId<id>> setContainer;
+    std::map<id, std::vector<StrongId<id>>::iterator> mapContainer;
```

Comment:
Is this really what we want? seems like it may be more hassle than its worth. As ben mentioned elsewhere, inserts etc will invalidate the iterators in the map which will be expensive to rebuild. I assume the reasoning here was to be more performant in a few operations that otherwise would require linear traversal but I'm not so sure that is needed? It appears that OrderedSet is O(n) for "array like" operations so I don't think we need to be super aggressive at optimizing at this point. Thoughts?